From 7d73939362abf0c8b574d7a03a4ebd5976d16d36 Mon Sep 17 00:00:00 2001
From: Quinn Jensen <quinn.jensen@freescale.com>
Date: Mon, 17 Nov 2008 13:31:45 -0700
Subject: [PATCH] ENGR00099227 MX25: Add I2C support

This patch increases the I2C bus clocks from 40 to 100 kHz,
to meet the 30 kbps requirement.

With 120 MHz perclk, the available nearest divisor (without going over)
is 1280 which produces 93.750 kbps.  This could probably be improved by
finding a better perclk and divisor combination.

Signed-off-by: Ross Wille <wille@freescale.com>
---
 arch/arm/mach-mx25/devices.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx25/devices.c b/arch/arm/mach-mx25/devices.c
index 4c50e5f..29dbc94 100644
--- a/arch/arm/mach-mx25/devices.c
+++ b/arch/arm/mach-mx25/devices.c
@@ -232,7 +232,7 @@ static struct resource mxci2c1_resources[] = {
 
 /*! Platform Data for MXC I2C */
 static struct mxc_i2c_platform_data mxci2c1_data = {
-	.i2c_clk = 40000,
+	.i2c_clk = 100000,
 };
 #endif
 
@@ -255,7 +255,7 @@ static struct resource mxci2c2_resources[] = {
 
 /*! Platform Data for MXC I2C */
 static struct mxc_i2c_platform_data mxci2c2_data = {
-	.i2c_clk = 40000,
+	.i2c_clk = 100000,
 };
 #endif
 
@@ -278,7 +278,7 @@ static struct resource mxci2c3_resources[] = {
 
 /*! Platform Data for MXC I2C */
 static struct mxc_i2c_platform_data mxci2c3_data = {
-	.i2c_clk = 40000,
+	.i2c_clk = 100000,
 };
 #endif
 
-- 
1.5.4.4

